direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaStreamElement.java

```diff
@@ -100,12 +107,13 @@ public StreamElement read(ConsumerRecord<String, byte[]> record, EntityDescripto
 
     @Override
     public ProducerRecord<String, byte[]> write(String topic, int partition, StreamElement data) {
-      @Nullable Iterable<Header> headers = null;
+      List<Header> headers = new ArrayList<>();
+      headers.add(
+          new RecordHeader(
+              KafkaAccessor.UUID_HEADER, data.getUuid().getBytes(StandardCharsets.UTF_8)));
```

Comment:
Hm. This is going to affect performance. Moreover, the UUID is hex-encoded, which is somewhat wasteful. Could we use a more compact represenation?

Reply:
I'm not for 100% sure if we can do that, because [StreamElement](https://github.com/datadrivencz/proxima-platform/blob/master/core/src/main/java/cz/o2/proxima/storage/StreamElement.java#L235) doesn't say `uuid` needs to be in UUID format, it's just an String and somebody can use it for own implementation which can be different. In that case we should also handle that exception with this "asString" fallback.

Reply:
We can do that, we can parse the UUID and use the compact representation only when the UUID is in UUID format. The question is if we should do that. I think that the overhead might be really significant in some situations. This can easily end up with something like doubling the size of each serialized element. With multiple consumers, this can substantially affect network throughput.
I think we could do dependency injection - instead of hardcoding the UUID serialization in the header, we can delegate it into a user-provided class (with two built-in implementations - a `NoOpUUIDSerializer` (this is the current) and a `RawHeaderUUIDSerialzer` (this is the one in this PR)). Users could then implement and override the serializer with their own.